### PR TITLE
KDTree: add support for custom distances

### PR DESCRIPTION
**Pull request by ayllon**
Need  Chebyshev Distance (`max(|x_i - y_i|)`) for computing the SED weights.

### CODE REVIEW

**Comment by ayllon** [comment]
@marcschefer sourcextractor does not use this implementation, so it would not be affected.
Should it, though?

**Comment by marcschefer** [comment]
Yeah, I made my own for SourceXtractor++

**Review by marcschefer — APPROVED** [review]

File: KdTree/KdTree/KdTree.h [hunk 1]
```diff
@@ -18,46 +18,112 @@
 #ifndef _SEUTILS_KDTREE_H_
```
Comment (marcschefer) [inline KdTree/KdTree/KdTree.h:18]: It should have an Alexandria guard, right? Otherwise we can get problems when compiling?

Comment (ayllon) [reply KdTree/KdTree/KdTree.h]: Good catch

**Comment by ayllon** [comment]
> Yeah, I made my own for SourceXtractor++

Well, this is a fork of yours :)

When we merged in sourcex I proposed to move the `KdTree` here because it could eventually be useful, and it has been! But I never got to modify sourcex to use this one.

**Comment by marcschefer** [comment]
I thought this was a different one until I saw the guard...

**Pull request merged**